Loop/View Controllers/SettingsTableViewController.swift

```diff
@@ -775,6 +783,35 @@ final class SettingsTableViewController: UITableViewController, IdentifiableClas
             self.dataManager.loopManager.settings.dosingEnabled = value
         }
     }
+    
+    private func saveTherapySetting(_ therapySetting: TherapySetting, _ therapySettings: TherapySettings) {
+        switch therapySetting {
+        case .glucoseTargetRange:
+            dataManager?.loopManager.settings.glucoseTargetRangeSchedule = therapySettings.glucoseTargetRangeSchedule
+        case .correctionRangeOverrides:
+            dataManager?.loopManager.settings.preMealTargetRange = therapySettings.preMealTargetRange
+            dataManager?.loopManager.settings.legacyWorkoutTargetRange = therapySettings.workoutTargetRange
+        case .suspendThreshold:
+            dataManager?.loopManager.settings.suspendThreshold = therapySettings.suspendThreshold
+        case .basalRate:
+            dataManager?.loopManager.basalRateSchedule = therapySettings.basalRateSchedule
+        case .deliveryLimits:
+            dataManager?.loopManager.settings.maximumBasalRatePerHour = therapySettings.maximumBasalRatePerHour
+            dataManager?.loopManager.settings.maximumBolus = therapySettings.maximumBolus
+        case .insulinModel:
+            if let insulinModel = therapySettings.insulinModel {
+                dataManager?.loopManager.insulinModelSettings = InsulinModelSettings(model: insulinModel as! InsulinModel)
```

Comment:
Curious what makes the `as!` necessary here; does `therapySettings.insulinModel` not always conform to InsulinModel?

Reply:
yikes, this is wrong... I'll fix it.